### PR TITLE
Adds startsWith and nstartsWith to Elastic Search

### PR DESCRIPTION
**Pull request by frzmohammadali**
### Summary of the changes 

Two filter operation handlers has been added for string operations with ElasticSearch:
- `startsWith`
- `nstartsWith`

Elastic search query to be used for these operations is the `PrefixQuery` under term-level type of queries.

Tests are available in `StringPrefixTests.cs` file. 

Feedbacks are more than welcome ;)

### CODE REVIEW

File: src/HotChocolate/ElasticSearch/test/Data.ElasticSearch.Tests/StringPrefixTests.cs [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+                .Field("test")
+                .UseFiltering<FooFilterType>()
+                .UseTestReport(Client)
+                .ResolveTestData(Client, _data))
+            .AddElasticSearchFiltering()
+            .BuildTestExecutorAsync();
+
+        const string query = @"
+        {
+            test(where: {and: [{bar: { startsWith: ""th"" }},{qux: { nstartsWith: ""do"" }}]}) {
```
Comment (PascalSenn) [inline src/HotChocolate/ElasticSearch/test/Data.ElasticSearch.Tests/StringPrefixTests.cs:232]: Can you check that this lines are not longer than 100 lines?

File: src/HotChocolate/ElasticSearch/test/Data.ElasticSearch.Tests/__snapshots__/StringPrefixTests.ElasticSearch_AndField.snap [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿{
```
Comment (PascalSenn) [inline src/HotChocolate/ElasticSearch/test/Data.ElasticSearch.Tests/__snapshots__/StringPrefixTests.ElasticSearch_AndField.snap:1]: Could it be that the query snapshot is missing for this one?

File: src/HotChocolate/ElasticSearch/test/Data.ElasticSearch.Tests/__snapshots__/StringPrefixTests.ElasticSearch_MultipleField.snap [hunk 3]
```diff
@@ -0,0 +1,12 @@
+﻿{
```
Comment (PascalSenn) [inline src/HotChocolate/ElasticSearch/test/Data.ElasticSearch.Tests/__snapshots__/StringPrefixTests.ElasticSearch_MultipleField.snap:1]: query?

File: src/HotChocolate/ElasticSearch/test/Data.ElasticSearch.Tests/__snapshots__/StringPrefixTests.ElasticSearch_MultipleField_OneNegated.snap [hunk 4]
```diff
@@ -0,0 +1,5 @@
+﻿{
```
Comment (PascalSenn) [inline src/HotChocolate/ElasticSearch/test/Data.ElasticSearch.Tests/__snapshots__/StringPrefixTests.ElasticSearch_MultipleField_OneNegated.snap:1]: query?

**Review by PascalSenn — CHANGES_REQUESTED** [review]
@frzmohammadali  Awesome work! This PR looks already really good :) 

A few things i noticed:
- We have a pretty strict 100 characters line length. Can you break the queries when possible?
- Could it be that there are some query snapshots missing or is there something wrong with the test helper?

File: src/HotChocolate/ElasticSearch/test/Data.ElasticSearch.Tests/StringPrefixTests.cs (hunk 1 above)
Comment (frzmohammadali) [reply src/HotChocolate/ElasticSearch/test/Data.ElasticSearch.Tests/StringPrefixTests.cs:232]: Could you elaborate? What exactly needs to be less than 100 lines? the test file you mean?

**Comment by frzmohammadali** [comment]
@PascalSenn Thanks a lot for your feedback. I'll take care of the points you mentioned and submit an update soon (this evening hopefully) :)

File: src/HotChocolate/ElasticSearch/test/Data.ElasticSearch.Tests/StringPrefixTests.cs (hunk 1 above)
Comment (frzmohammadali) [reply src/HotChocolate/ElasticSearch/test/Data.ElasticSearch.Tests/StringPrefixTests.cs]: Thanks for letting me know about this policy. Luckily every line were less than 100 characters already, but I made some line breaks here and there to ensure this and increase readability.

File: src/HotChocolate/ElasticSearch/test/Data.ElasticSearch.Tests/__snapshots__/StringPrefixTests.ElasticSearch_AndField_WithNegation_query.snap [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+      "filter": [
+        {
+          "bool": {
+            "must": [
+              {
+                "prefix": {
+                  "bar": {
+                    "value": "th"
+                  }
+                }
```
[the remark this answers is not in the document]
Comment (frzmohammadali) [reply src/HotChocolate/ElasticSearch/test/Data.ElasticSearch.Tests/__snapshots__/StringPrefixTests.ElasticSearch_AndField_WithNegation_query.snap:13]: As far as prefix queries are concerned, nothing was mentioned about wildcards in the elastic query documentation to the best of my knowledge. I will try it myself though, and let you know about the result. Currently I have an issue with my local elasticsearch docker container (created by squadron) which I did not have yesterday :) I look into this case once I resolved my container issue.

File: src/HotChocolate/ElasticSearch/test/Data.ElasticSearch.Tests/__snapshots__/StringPrefixTests.ElasticSearch_AndField.snap (hunk 2 above)
Comment (frzmohammadali) [reply src/HotChocolate/ElasticSearch/test/Data.ElasticSearch.Tests/__snapshots__/StringPrefixTests.ElasticSearch_AndField.snap:1]: It should be found in `__snapshots__/StringPrefixTests.ElasticSearch_AndField_query.snap` and it is checked to the repository. Could you double check that you don't have such file?

File: src/HotChocolate/ElasticSearch/test/Data.ElasticSearch.Tests/__snapshots__/StringPrefixTests.ElasticSearch_MultipleField.snap (hunk 3 above)
Comment (frzmohammadali) [reply src/HotChocolate/ElasticSearch/test/Data.ElasticSearch.Tests/__snapshots__/StringPrefixTests.ElasticSearch_MultipleField.snap:1]: Similarly, the query for this test should be found under `__snapshots__/StringPrefixTests.ElasticSearch_MultipleField_query.snap`.

File: src/HotChocolate/ElasticSearch/test/Data.ElasticSearch.Tests/__snapshots__/StringPrefixTests.ElasticSearch_MultipleField_OneNegated.snap (hunk 4 above)
Comment (frzmohammadali) [reply src/HotChocolate/ElasticSearch/test/Data.ElasticSearch.Tests/__snapshots__/StringPrefixTests.ElasticSearch_MultipleField_OneNegated.snap:1]: should be found under `__snapshots__/StringPrefixTests.ElasticSearch_MultipleField_OneNegated_query.snap`.

File: src/HotChocolate/ElasticSearch/test/Data.ElasticSearch.Tests/__snapshots__/StringPrefixTests.ElasticSearch_MultipleField.snap (hunk 3 above)
Comment (PascalSenn) [reply src/HotChocolate/ElasticSearch/test/Data.ElasticSearch.Tests/__snapshots__/StringPrefixTests.ElasticSearch_MultipleField.snap:1]: Cool :) Just overlooked it apparently :)

File: src/HotChocolate/ElasticSearch/test/Data.ElasticSearch.Tests/__snapshots__/StringPrefixTests.ElasticSearch_AndField_WithNegation_query.snap (hunk 5 above)
[the remark this answers is not in the document]
Comment (frzmohammadali) [reply src/HotChocolate/ElasticSearch/test/Data.ElasticSearch.Tests/__snapshots__/StringPrefixTests.ElasticSearch_AndField_WithNegation_query.snap:13]: After more trying and testing, I can confirm that "*" in the prefix query does not have any effect as wildcard and is treated as a normal character.

**Review by PascalSenn — APPROVED** [review]
Very well done ali!

**Pull request merged**